pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -93,12 +93,20 @@ protected void processTable(String tableNameWithType) {
 
     // Delete segments based on segment lineage and clean up segment lineage metadata.
     manageSegmentLineageCleanupForTable(tableConfig);
+
+    // call segment deletion manager for that table
+    _pinotHelixResourceManager.getSegmentDeletionManager().removeAgedDeletedSegments(tableConfig);
   }
 
   @Override
   protected void postprocess() {
-    LOGGER.info("Removing aged (more than {} days) deleted segments for all tables", _deletedSegmentsRetentionInDays);
-    _pinotHelixResourceManager.getSegmentDeletionManager().removeAgedDeletedSegments(_deletedSegmentsRetentionInDays);
+    LOGGER.info("Clean up aged (more than {} days) deleted segments for all tables", _deletedSegmentsRetentionInDays);
+    List<String> tableWithAgedDeletedSegments =
+        _pinotHelixResourceManager.getSegmentDeletionManager().getTableWithAgedDeletedSegments();
+    for (String tableName : tableWithAgedDeletedSegments) {
+      TableConfig tableConfig = _pinotHelixResourceManager.getTableConfig(tableName);
```

Comment:
make this call parsing and only pass in the retentionMs to SegmentDeletionManager.